### PR TITLE
docs/bpf_attach: add documentation for bpf_attach event

### PR DESCRIPTION
**Pull request by roikol**
## Initial Checklist

- [x] There is an issue describing the need for this PR.
- [x] Git log contains summary of the change.
- [x] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

```
Author: RoiKol <roi.kol@aquasec.com>
Date:   Thu Dec 15 14:46:18 2022 +0200

    docs/bpf_attach: add documentation for bpf_attach event
```

Fixes: #2308

## Type of change

- [ ] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [ ] Code refactor (code improvement and/or code removal)
- [x] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## How Has This Been Tested?

Tests being included in this PR:

- [ ] Test File A
- [ ] Test File B

Reproduce the test by running:

- command 01
- command 02

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [x] I have made corresponding changes to the documentation.
- [x] My code follows the style guidelines (C and Go) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented all functions/methods created explaining what they do.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix, or feature, is effective.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [x] Subject starts with "subsystem|file: description".
- [x] Do not end the subject line with a period.
- [x] Limit the subject line to 50 characters.
- [x] Separate subject from body with a blank line.
- [x] Use the imperative mood in the subject line.
- [x] Wrap the body at 72 characters.
- [x] Use the body to explain what and why instead of how.

### CODE REVIEW

File: docs/contributing/events/security_alerts/bpf_attach.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+## Description
+An event marking that a BPF program was attached to a probe in the system.
+It occurs whenever a BPF program is attached to a perf event of the types: kprobe, uprobe or tracepoint.
+The purpose of the event is to give the user information about the BPF program, 
+as well as information about the probe itself.
+
+### Note
+The event doesn't occur for FIFO files or other files with no persistent offsets,
+to reduce spam events.
```
Comment (yanivagman) [inline docs/contributing/events/security_alerts/bpf_attach.md:14]: It is not clear what does this mean, and how it is related to this event

File: docs/contributing/events/security_alerts/bpf_attach.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+## Arguments
+* `prog_type`:`int`[K] - the BPF program type.
+* `prog_name`:`const char*`[K] - the BPF program type (first 16 bytes only, as this is how it is saved in the kernel).
+* `perf_symbol`:`const char*`[K] - name/path of the symbol the BPF program is being attached to.
+* `perf_addr`:`u64`[K] - address/offset of the symbol the BPF program is being attached to.
+* `prog_write_user`:`int`[K] - whether of not the BPF program uses the bpf_probe_write_user() helper function.
```
Comment (yanivagman) [inline docs/contributing/events/security_alerts/bpf_attach.md:21]: typo: whether **or** not

File: docs/contributing/events/security_alerts/bpf_attach.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+kprobe
+#### Purpose
+check whether the BPF program uses the bpf_probe_write_user() helper function
+
+## Example Use Case
+./tracee-ebpf -t e=bpf_attach
+
+## Issues
+the 'check_helper_call' and 'check_map_func_compatibility' serves the same purpose. 
+in some kernels one of this symbols would not exist - therefore libbpf will output an error (execution will continue secceffuly due to the other hook).
```
Comment (yanivagman) [inline docs/contributing/events/security_alerts/bpf_attach.md:60]: typo successfully

**Review by yanivagman — CHANGES_REQUESTED** [review]

File: docs/contributing/events/security_alerts/bpf_attach.md (hunk 1 above)
Comment (roikol) [reply docs/contributing/events/security_alerts/bpf_attach.md:14]: sorry, it is not related (left over from copy-paste).
should i keep the `### Note` section empty? or remove it?

**Review by yanivagman — APPROVED** [review]
LGTM

**Pull request merged**